### ISSUE TITLE
Checkbox in latest React deploy does not have mouse interactability

### ISSUE THREAD

**Issue by jendowns** [carbon-design-system/carbon#6048]
## Detailed description

Check out the latest React deploy here: https://react.carbondesignsystem.com/?path=/story/checkbox--checked

In the latest Firefox and Chrome, I am unable to interact with the `Checkbox` using my mouse; I cannot check/uncheck it. 

With my keyboard, I *can* interact with the `Checkbox` in that deployment.

Looking at the git history, I think this comes from the following PR: https://github.com/carbon-design-system/carbon/pull/5373

Specifically, changes made to `Checkbox.js` that replaced the `htmlFor` attribute on the `label` with an `id` attribute. Form elements generally need an `id` on the `input` element, and then a `for` element on the corresponding label -- since this is React, we would need to use `htmlFor` here: https://reactjs.org/docs/dom-elements.html#htmlfor

So to summarize, I believe restoring `htmlFor` in this case would resolve the issue.

**Comment by jendowns** [comment]
I will have a fix for this momentarily 👍

**Comment by dakahn** [comment]
This is what I get for _only checking my work with keyboard/screen reader_ 🤦. Thanks so much for catching this

**Comment by jendowns** [comment]
No worries at all @dakahn! Thank you for the fast response. 🎉
